src/wp-includes/blocks.php

```diff
@@ -40,6 +40,221 @@ function unregister_block_type( $name ) {
+ * automatically generated handle name. It returns unprocessed script handle
+ * otherwise.
+ *
+ * @since 5.5.0
+ *
+ * @param array  $metadata Block metadata.
+ * @param string $field_name Field name to pick from metadata.
+ *
+ * @return string|boolean Script handle provided directly or created through
+ *     script's registration, or false on failure.
+ */
+function register_block_script_handle( $metadata, $field_name ) {
+	if ( empty( $metadata[ $field_name ] ) ) {
+		return false;
+	}
+	$script_handle = $metadata[ $field_name ];
+	$script_path   = remove_block_asset_path_prefix( $metadata[ $field_name ] );
+	if ( $script_handle === $script_path ) {
+		return $script_handle;
+	}
+
+	$script_handle     = generate_block_asset_handle( $metadata['name'], $field_name );
+	$script_asset_path = realpath(
+		dirname( $metadata['file'] ) . '/' .
+		substr_replace( $script_path, '.asset.php', - strlen( '.js' ) )
+	);
+	if ( ! file_exists( $script_asset_path ) ) {
+		$message = sprintf(
+			/* translators: %1: field name. %2: block name */
+			__( 'The asset file for the "%1$s" defined in "%2$s" block definition is missing.', 'default' ),
+			$field_name,
+			$metadata['name']
+		);
+		_doing_it_wrong( __FUNCTION__, $message, '5.5.0' );
+		return false;
+	}
+	$script_asset = require( $script_asset_path );
+	$result       = wp_register_script(
+		$script_handle,
+		plugins_url( $script_path, $metadata['file'] ),
+		$script_asset['dependencies'],
+		$script_asset['version']
+	);
+	return $result ? $script_handle : false;
+}
+
+/**
+ * Finds a style handle for the block metadata field. It detects when a path
+ * to file was provided and registers the style under automatically
+ * generated handle name. It returns unprocessed style handle otherwise.
+ *
+ * @since 5.5.0
+ *
+ * @param array  $metadata Block metadata.
+ * @param string $field_name Field name to pick from metadata.
+ *
+ * @return string|boolean Style handle provided directly or created through
+ *     style's registration, or false on failure.
+ */
+function register_block_style_handle( $metadata, $field_name ) {
+	if ( empty( $metadata[ $field_name ] ) ) {
+		return false;
+	}
+	$style_handle = $metadata[ $field_name ];
+	$style_path   = remove_block_asset_path_prefix( $metadata[ $field_name ] );
+	if ( $style_handle === $style_path ) {
+		return $style_handle;
+	}
+
+	$style_handle = generate_block_asset_handle( $metadata['name'], $field_name );
+	$block_dir    = dirname( $metadata['file'] );
+	$result       = wp_register_style(
+		$style_handle,
+		plugins_url( $style_path, $metadata['file'] ),
+		array(),
+		filemtime( realpath( "$block_dir/$style_path" ) )
+	);
+	return $result ? $style_handle : false;
+}
+
+/**
+ * Registers a block type from metadata stored in the `block.json` file.
+ *
+ * @since 5.5.0
+ *
+ * @param string $file_or_folder Path to the JSON file with metadata definition for
+ *     the block or path to the folder where the `block.json` file is located.
```

Comment:
```suggestion
 * @param string $file_or_folder Path to the JSON file with metadata definition for
 *                               the block or path to the folder where the `block.json` file is located.
```

In case of a line break the indentation should match the previous line.